From 3fb1a1832494efe2488587c6d228be0f692299ba Mon Sep 17 00:00:00 2001
From: Jayesh Kumar Tank <jayesh.kumarx.tank@intel.com>
Date: Thu, 8 Mar 2018 15:21:19 -0800
Subject: [PATCH 2/2] [DRI3] Hooks to enable DRI3 and Present

Enable DRI3 and Present Extension support, mark dependencies.

Usage     : '--enable-dri3' config flag to use this feature.
Dependency: libva should be build with DRI3

Signed-off-by: Jayesh Kumar Tank <jayesh.kumarx.tank@intel.com>
---
 configure.ac    | 28 ++++++++++++++++++++++++++++
 src/Makefile.am |  3 +++
 2 files changed, 31 insertions(+)

diff --git a/configure.ac b/configure.ac
index 795d36a..6736124 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,6 +52,11 @@ AC_ARG_ENABLE(x11,
                     [build with VA/X11 API support @<:@default=yes@:>@])],
     [], [enable_x11="yes"])
 
+AC_ARG_ENABLE(dri3,
+    [AC_HELP_STRING([--enable-dri3],
+                    [build with VA/X11-DRI3 API support @<:@default=no@:>@])],
+    [], [enable_dri3="no"])
+
 AC_ARG_ENABLE([wayland],
     [AC_HELP_STRING([--enable-wayland],
                     [build with VA/Wayland API support @<:@default=yes@:>@])],
@@ -135,6 +140,28 @@ if test "$USE_X11" = "yes"; then
 fi
 AM_CONDITIONAL(USE_X11, test "$USE_X11" = "yes")
 
+dnl Check for DRI3
+USE_DRI3="$enable_dri3"
+
+if test "$USE_X11:$enable_dri3" = "no:yes"; then
+    AC_MSG_ERROR([VA/X11-DRI3 explicitly enabled, but VA/X11 isn't built])
+fi
+
+if test "$enable_dri3" != "no"; then
+    PKG_CHECK_MODULES([DRI3], [xcb x11-xcb xcb-dri3 \
+       xcb-sync xshmfence xcb-present], [USE_DRI3="yes"], [:])
+
+    if test "x$USE_DRI3" = "xno" -a "x$enable_dri3" = "xyes"; then
+       AC_MSG_ERROR([VA/X11-dri3 explicitly enabled, however $DRI3_PKG_ERRORS])
+    fi
+
+    if test "$USE_DRI3" = "yes"; then
+       AC_DEFINE([HAVE_VA_DRI3], [1], [Defined to 1 if VA/X11-DRI3 API is
+        enabled])
+    fi
+fi
+AM_CONDITIONAL(USE_DRI3, test "$USE_DRI3" = "yes")
+
 dnl Check for VA-API drivers path
 AC_ARG_VAR(LIBVA_DRIVERS_PATH, [drivers install path])
 if test -z "$LIBVA_DRIVERS_PATH"; then
@@ -197,6 +224,7 @@ AC_OUTPUT([
 dnl Print summary
 BACKENDS="drm"
 AS_IF([test "$USE_X11" = "yes"], [BACKENDS="$BACKENDS x11"])
+AS_IF([test "$USE_DRI3" = "yes"], [BACKENDS="$BACKENDS x11-dri3"])
 AS_IF([test "$USE_WAYLAND" = "yes"], [BACKENDS="$BACKENDS wayland"])
 
 echo
diff --git a/src/Makefile.am b/src/Makefile.am
index c26293d..ac8c8cb 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -48,6 +48,9 @@ driver_libs = \
 	-lpthread -lm -ldl	\
 	$(DRM_LIBS)		\
 	$(NULL)
+if USE_DRI3
+driver_libs     += $(DRI3_LIBS)
+endif
 
 include Makefile.sources
 
-- 
2.17.0

